components/forms/w-textarea.vue

```diff
@@ -7,6 +7,16 @@ import { default as wField, fieldProps } from './w-field.vue';
 const p = defineProps(fieldProps);
 const emit = defineEmits(['update:modelValue']);
 const model = createModel({ props: p, emit });
+
+const textAreaClasses = (hasValidationErrors) => [
+  ccInput.base,
+  ccInput.textArea,
+  !!p.placeholder && ccInput.placeholder,
+  !hasValidationErrors && !p.disabled && !p.readOnly && ccInput.default,
+  hasValidationErrors && !p.disabled && !p.readOnly && ccInput.invalid,
+  !hasValidationErrors && p.disabled && !p.readOnly && ccInput.disabled,
+  !hasValidationErrors && !p.disabled && p.readOnly && ccInput.readOnly,
+];
```

Comment:
Just a shot in the dark because I'm not sure - doesn't this need to be `computed`?

Reply:
Yes I also wanted to change it to a computed, but then I got this error:  `"(in promise) TypeError: $setup.textAreaClasses is not a function"` 🙈 

Since `textAreaClasses` depends on `hasValidationErrors`, which is provided by the slot scope (`#default`) of the `w-field` component, `hasValidationErrors` is only available in the template and not directly inside the `setup` function where computed properties are typically defined. `textAreaClasses` is dependent on `hasValidationErrors`, which is already dynamically updated each time the slot content is rendered. So I think, in this case, it is better to use a function instead of a computed property. This way, the code is simpler and aligns better with the data flow from the slot to the template. :)

Reply:
I understand now! Thanks for explaining :)